model-optimizer/extensions/front/tf/SSliceComplex.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+import logging as log
+
+from mo.front.common.replacement import FrontReplacementSubgraph
+from mo.front.subgraph_matcher import SubgraphMatch
+from mo.graph.graph import Graph
+
+
+class SSliceComplex(FrontReplacementSubgraph):
+    """
+    Some TF models contain the sub-graph
+               SomeOp
+                 |
+    --------------------------
+    |                        |
+    StridedSlice          StridedSlice
+    |                       |
+    ------------------------
+         Complex
+          |
+          |       other inputs
+          |       |  ...  |
+         -------------------
+                 SomeOp1
+
+    Here SomeOp is some node with real output and with the shape [N_0, ..., N_{r - 1}, 2], and StridedSlice nodes
+    have output shapes [N_0, ..., N_{r - 1}].
+
+    But MO and Inference Engine do not support for complex tensors. Hence, we need to replace this sub-graph with
+
+         SomeOp   other inputs
+          |       |  ...  |
+         -------------------
+                 SomeOp1
+
+    And, in such replacement, we should mark SomeOp1 as needed to correct. For example, if SomeOp1 is Roll,
+    we need to correct axes of Roll, because input data are real now.
```

Comment:
After this transformation we need to mark SomeOp1 operation that its input rank has changed because its inputs/attributes should probably be updated. Currently we have such a case for a Roll operation.

Reply:
Fixed.